doc/osmp_test_architecture.md

```diff
@@ -0,0 +1,224 @@
+
+### CL 1.1: Linter
+
+These tests consist of linters checking the formatting against style guides for code, markdown and xml.
+
+#### C++ Linter
+
+C++ code is checked with clang according to the clang-format and clang-tidy files in the repository.
+It is mandatory for every code repository to provide these config files.
+As a best practice, [clang-format-all](https://github.com/eklitzke/clang-format-all) can be used to automatically format all C++ files in the repository.
+Be aware, that formatting might break functionality, e.g. by a different order of includes.
+Be sure to check after formatting.
+
+An implementation of the corresponding GitHub action can be found in the [sensor model template repository](https://github.com/openMSL/sl-1-0-sensor-model-repository-template/tree/main/.github/workflows/cpp-linter.yml).
+
+#### Markdown Linter
+
+Markdown files are checked by a markdown linter according to [these](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md) basic rules.
+
+An implementation of the corresponding GitHub action can be found in the [sensor model template repository](https://github.com/openMSL/sl-1-0-sensor-model-repository-template/tree/main/.github/workflows/markdown.yml).
+It also contains an example [markdownlint.json](https://github.com/openMSL/sl-1-0-sensor-model-repository-template/tree/main/.github/workflows/markdownlint.json) specifying variations of the applied rules.
+
+#### SRMD Validator
+
+Every model repository shall include an [SRMD file](https://pmsfit.github.io/SSPTraceability/master/#_srmd) conforming to the [SSP tracability](https://pmsfit.github.io/SSPTraceability/master/) process.
+This file is checked against the [SRMD XML schema](https://github.com/PMSFIT/SSPTraceability).
+
+### CL 1.2: Build
+
+These tests check if the model can be build and the built FMU is compliant with the [FMI standard](https://github.com/modelica/fmi-standard).
+
+#### Cmake
+
+This test checks if cmake runs without errors and the model can be built with all necessary dependencies, such as OSI with Protobuf.
+
+An implementation of the corresponding GitHub action can be found in the [sensor model template repository](https://github.com/openMSL/sl-1-0-sensor-model-repository-template/tree/main/.github/workflows/build.yml).
+
+#### FMU Compliance Checker
+
+The official [FMU Compliance Checker](https://github.com/modelica-tools/FMUComplianceChecker) by Modelica is used to check if the built FMU is compliant with the [FMI standard](https://github.com/modelica/fmi-standard).
+This includes checking the binaries as well as included xml files such as the modelDescription.
+
+Since the FMU Compliance Checker does not seem to be further maintained, [FMPy](https://github.com/CATIA-Systems/FMPy) is used as a second validation tool.
+
+### CL 1.3 Unit Tests
+
+Models shall include individual unit tests.
```

Comment:
I think it is a good idea to make a remark somewhere similar like this and write these RFC keywords in upper letters "SHALL"

> Many of the terms in this standard are derived from [The Dublin Core, RFC 2413][3].
> 
> The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
> “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
> interpreted as described in [RFC 2119][27].

Reply:
Since this is a test architecture description and not a requirements document, I am not sure if we should emphasize it that way. What do you think? Are the other sections in the document, where you would emphasize these keywords?

Reply:
done.